spec/lib/dfe_sign_in_api_spec.rb

```diff
@@ -0,0 +1,87 @@
+require 'rails_helper'
+require 'dfe_sign_in_api'
+
+RSpec.describe DFESignIn::API do
+  describe '#users' do
+    context 'when the response code is 200' do
+      it 'returns the users from the API' do
+        stub_api_response_for_page(1)
+        response = described_class.new.users
+
+        expect(response).to eq(JSON.parse(response_file(1)))
+        expect(response['page']).to eq(1)
+        expect(response['numberOfPages']).to eq(2)
+      end
+
+      it 'returns the users from the API for a given page ' do
+        stub_api_response_for_page(2)
+        response = described_class.new.users(page: 2)
+
+        expect(response).to eq(JSON.parse(response_file(2)))
+        expect(response['page']).to eq(2)
+        expect(response['numberOfPages']).to eq(2)
+      end
+
+      it 'sets a token in the header of the request' do
+        Timecop.freeze(Time.zone.now) do
+          payload = {
+            iss: 'schooljobs',
+            exp: (Time.now.getlocal + 60).to_i,
+            aud: 'signin.education.gov.uk'
+          }
+
+          expected_token = JWT.encode(payload, DFE_SIGN_IN_PASSWORD, 'HS256')
+
+          stub_api_response_for_page(2)
+          described_class.new.users(page: 2)
+
+          expect(a_request(:get, "#{DFE_SIGN_IN_URL}/users?page=2&pageSize=25")
+            .with(headers: { 'Authorization' => "Bearer #{expected_token}" }))
+            .to have_been_made
+        end
+      end
+    end
+
+    context 'when the external response status is 500' do
+      before { stub_api_response_with_external_error(1) }
+
+      it 'raises an external server error' do
+        expect { described_class.new.users }.to raise_error(DFESignIn::ExternalServerError)
+      end
+    end
+
+    context 'when the response was 403' do
+      it 'returns a JSON with success equal to false' do
```

Comment:
Looks like it raises an error as opposed to what you describe,

Reply:
Thanks for spotting this. Changed the expectation, forgot the description.